src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -964,12 +964,17 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
                 var constantValueDiagnostics = localSymbol.GetConstantValueDiagnostics(initializerOpt);
                 foreach (var diagnostic in constantValueDiagnostics)
                 {
-                    diagnostics.Add(diagnostic);
-                    hasErrors = true;
+                    localDiagnostics.Add(diagnostic);
+                    if (diagnostic.Severity == DiagnosticSeverity.Error)
+                    {
+                        hasErrors = true;
+                    }
                 }
             }
 
-            diagnostics.AddRangeAndFree(localDiagnostics);
+            diagnostics.AddRange(localDiagnostics.AsEnumerable().Distinct());
```

Comment:
@jcouv I have reverted the changes that were the reason I added this.

There are still cases where duplicate warnings will be produced and I think those should be deduped, but I'll create a separate issue for that.